src/PrestaShopBundle/Controller/Admin/FrameworkBundleAdminController.php

```diff
@@ -524,4 +526,16 @@ protected function getErrorMessageForException(Exception $e, array $messages)
             $e->getMessage()
         );
     }
+
+    protected function render($view, array $parameters = [], Response $response = null)
+    {
+        /** @var HookDispatcherInterface $hookDispatcher */
+        $hookDispatcher = $this->get('prestashop.core.hook.dispatcher');
+        /** @var RenderedHook $renderedHook */
+        $renderedHook = $hookDispatcher->dispatchRenderingWithParameters('actionBackOfficeControllerSetVariables');
+
+        $parameters = array_merge($renderedHook->getContent(), $parameters);
```

Comment:
```suggestion
        $parameters = array_merge($parameters, $renderedHook->getContent());
```
I'm not sure but I think the idea is to allow the modules override the core variables, if so then the hook content must be the final argument of array merge

Reply:
As I understood the idea is to add extra variable to the twig along with the existing ones (passed from Controller)

Reply:
Mmmm ... indeed @jolelievre's idea is good. And we could even pass `$parameters` to the hook to allow it to enrich the data or compute data from it !

Reply:
also have a small comment at the bottom about the structure the variables are returned. If we take @jolelievre's idea then we need some changes in the code I guess. Separated parsing for array and string case;